### PR TITLE
[ASCII-1481] Catch UNEXPECTED ERROR in e2e diagnose

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?
This PR updates diagnose e2e test to catch UNEXPECTED ERROR.
 

### Motivation
Catch https://github.com/DataDog/datadog-agent/pull/24604.

### CODE REVIEW

File: test/new-e2e/tests/agent-subcommands/diagnose/diagnose_common_test.go [hunk 1]
```diff
@@ -117,6 +117,11 @@ func (v *baseDiagnoseSuite) TestDiagnoseVerbose() {
 	assert.Contains(v.T(), diagnose, "connectivity-datadog-core-endpoints")
 }
 
+func (v *baseDiagnoseSuite) AssertOutputNotError(diagnose string) {
+	assert.NotContains(v.T(), diagnose, "FAIL")
```
Comment (louis-cqrl) [inline test/new-e2e/tests/agent-subcommands/diagnose/diagnose_common_test.go:121]: In the future, if we want to add other key word to exclude this format can be nice too ?
really not mandatory was just a thought.
for _, forbidden := range []string{"FAIL", "UNEXPECTED ERROR"} {
    assert.NotContains(v.T(), diagnose, forbidden)
}

**Review by louis-cqrl — APPROVED** [review]
Nice PR

File: test/new-e2e/tests/agent-subcommands/diagnose/diagnose_common_test.go (hunk 1 above)
Comment (ogaca-dd) [reply test/new-e2e/tests/agent-subcommands/diagnose/diagnose_common_test.go:121]: I think for two it is OK to not use a loop but for more it is definitely a good idea.

**Pull request merged**